overlord/snapshotstate/backend/backend.go

```diff
@@ -453,7 +459,13 @@ func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer,
 	if err := osutil.RunWithContext(ctx, cmd); err != nil {
 		matches, count := matchCounter.Matches()
 		if count > 0 {
-			return fmt.Errorf("cannot create archive: %s (and %d earlier)", matches[len(matches)-1], count-1)
+			note := ""
+			if len(matches) > 5 {
+				matches = matches[len(matches)-5:]
```

Comment:
Isn't MatchCounter taking core of only keeping up to N matches?

Reply:
Silly me, yes that's a fair point.